### ISSUE TITLE
Too many slides make Slick hardly working in Safari

### ISSUE THREAD

**Issue by silllli** [kenwheeler/slick#1986]
I am sorry if this issue has already been addressed, I could not find anything about it.

I have a pretty large carousel in terms of file size: about 60 image slides with a total size of 25 mb. No problem for Chrome and Firefox, but in Safari Slick becomes absolutely unusable. Everything is lagging, sometimes up to several seconds.

This is happening on Mac OS X, El Capitan running, Safari 9.0.2.

**Comment by gitsilvan** [comment]
OK I just tested some more: it's also happening with smaller images and less slides - 15 is alright while 30 is already too much.

There is no problem when using Slick 1.4.1 (!).

**Comment by kenwheeler** [comment]
can you reproduce this in a jsfiddle?

**Comment by gitsilvan** [comment]
Yes: https://jsfiddle.net/zostaqko/.

Chrome and Firefox work as expected, Safari is lagging, barely smooth transitions visible.

**Comment by rahulgaba16** [comment]
Use version 1.5.8. latest version has some problems with ios
do npm install slick-carousel@1.5.8

**Comment by gitsilvan** [comment]
This does not help. As I have written is slick 1.4.1 the first working version.

**Comment by tomaszpoliszuk** [comment]
From what i observed loading time of images from https://placeimg.com/ is long in general.

Also loading 25mb of webpage is not wise move - consider using lazy loading.

What should be tested:
- images from source other than https://placeimg.com/ with current version of slick for big images ~25mb or more
- same as above using lazy loading

**Comment by gitsilvan** [comment]
Yes, loading from placeimg.com is slow and it is not clever to load 25 mb of data for the website – but in my case it is the only solution that makes sense. Of course I the images I used in my local development did not come from placeimg.com, but the client's ones, with the same problems.

As I have stated: this is browser depending and working fine in Chrome and Firefox.

**Comment by tomaszpoliszuk** [comment]
Yeah - i get it - tested on safari for now (it was more like note to self but with option for others to do this before i will have time;).
On the evening i will have some time to test this.

**Comment by tomaszpoliszuk** [comment]
changing to lazy loading helps a little but only for first time - on second first slides are glitchy - last ones are ok:
https://jsfiddle.net/zostaqko/12/
the same (first slides are glitchy - last ones are ok) seems to happen without lazy loading https://jsfiddle.net/zostaqko/6/

**Comment by lecollective** [comment]
Did you ever resolve this issue? Experiencing the same

**Comment by digimattress** [comment]
Running into this also in 1.9.0 and the Edge browser. I had to go back to 1.4.1 before it would load in under 5 minutes (compared to 2 seconds in Chrome) with 40 slides (even with no images). Lazyloading made no difference.
